packages/components/src/Button/button.stories.tsx

```diff
@@ -1,8 +1,28 @@
-import Button from "./button";
+import Button, { ButtonProps } from "./button";
+import Clickable, { ClickableProps } from "../Clickable";
 import CheckCircleRoundedIcon from "../Icons/CheckCircleRounded";
+import Heading from "../Heading";
 import React from "react";
 import { Story } from "@storybook/react/types-6-0";
 import Text from "../Text";
+import styles from "./button.stories.module.css";
+
+const sizes: Array<ButtonProps["size"]> = ["small", "medium", "large"];
+const allColors: Array<ButtonProps["color"]> = [
+  "alert",
+  "brand",
+  "neutral",
+  "success",
+  "warning",
+];
+const variants: Array<ButtonProps["variant"]> = ["flat", "outline", "link"];
+const states: Array<ClickableProps<"button">["state"] | "disabled"> = [
+  "inactive",
+  "hover",
+  "focus",
+  "active",
+  "disabled",
+];
```

Comment:
Can we get rid of the explicit typing with something like

```ts
const states = [
  // ... items
] as const;
```

My thought process is this should result in a "tuple" type, where typescript understands what specific value each item in the array is.

As we iterate that array, we should end up with a type that TS lets us pass to the component

```ts
states.map((state) => {
  // TypeScript should understand here that state has the type of
  // 'inactive' | 'hover' | 'focus' | etc... , which we pass to `state`.
  return <SomeComponent state={state} />;
});
```

[Can see something like this working in the TS playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAysCGwIGcoF4oHIBmB7XmUAPlgEbwBOhJm5AXpgNwBQz2ArgHYDGwAlrk5QkyYAApRiCAC5YCEQEooAbwC+rboNFRR7bNnRQA2jnyYANGUoWrDALpR4qTZ1Etmu-QDoAtvDBiYsAAFnycAOZKaAB8KsxQCcIo4iFhkSyqCoxAA).

Hoping that works for all of these consts.

Reply:
will try!